### PR TITLE
Improve 'Send Test Email'

### PR DESCRIPTION
**Pull request by ultramegatom**
add option to choose a document
add option to add parameters

add from field (prefilled with system settings)

add send test mail button to email and newsletter documents (will prefill the email window with data)

### CODE REVIEW

**Comment by ultramegatom** [comment]
this is how it looks:

image

File: bundles/AdminBundle/Resources/public/js/pimcore/helpers.js [hunk 1]
```diff
@@ -1628,60 +1628,195 @@ pimcore.helpers.searchAndMove = function (parentId, callback, type) {
...
+            {name: 'key', type: 'string'},
+            {name: 'value', type: 'string'}
+        ]
+    });
+
+    var paramGrid = Ext.create('Ext.grid.Panel', {
+        store: paramsStore,
+        columns: [
+            {
+                text: 'key',
```
Comment (brusch) [inline bundles/AdminBundle/Resources/public/js/pimcore/helpers.js]: we should use a a translation key here: `t("key")`

File: bundles/AdminBundle/Resources/public/js/pimcore/helpers.js [hunk 2]
```diff
@@ -1628,60 +1628,195 @@ pimcore.helpers.searchAndMove = function (parentId, callback, type) {
...
+                dataIndex: 'key',
+                editor: new Ext.form.TextField(),
+                width: 200
+            },
+            {
+                text: 'value',
```
Comment (brusch) [inline bundles/AdminBundle/Resources/public/js/pimcore/helpers.js]: same here: `t("value")`

File: bundles/AdminBundle/Controller/Admin/EmailController.php [hunk 3]
```diff
@@ -346,14 +346,37 @@ public function sendTestEmailAction(Request $request)
...
+        }
         $mail->addTo($request->get('to'));
         $mail->setSubject($request->get('subject'));
         $mail->setIgnoreDebugMode(true);
 
-        if ($request->get('type') == 'text') {
+        if ($request->get('emailType') == 'text') {
             $mail->setBodyText($request->get('content'));
-        } else {
+        } elseif($request->get('emailType') == 'text') {
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/EmailController.php]: This should be `} elseif($request->get('emailType') == 'html') {` right?

File: bundles/AdminBundle/Resources/public/js/pimcore/helpers.js (hunk 1 above)
Comment (ultramegatom) [reply bundles/AdminBundle/Resources/public/js/pimcore/helpers.js]: done

File: bundles/AdminBundle/Resources/public/js/pimcore/helpers.js (hunk 2 above)
Comment (ultramegatom) [reply bundles/AdminBundle/Resources/public/js/pimcore/helpers.js]: done

File: bundles/AdminBundle/Controller/Admin/EmailController.php (hunk 3 above)
Comment (ultramegatom) [reply bundles/AdminBundle/Controller/Admin/EmailController.php]: good catch :)

done

**Comment by brusch** [comment]
Thanks!

**Pull request merged**